### PR TITLE
Video structure update

### PR DESCRIPTION
**Pull request by mjjcha**
ezgif com-gif-maker (25)

- Update home page to have a video background
- Add GenericVideoPlayer component to play both Youtube and Dailymotion types
- Add LoadingState component
- Update `members` => `team_members`
- Update `youtube_mappings` => `video_mappings`
- Add `members` to represent each member
- General styling improvements

### CODE REVIEW

File: client/public/index.html [hunk 1]
```diff
@@ -33,16 +33,24 @@
...
-      The build step will place the bundled scripts into the <body> tag.
-
-      To begin the development, run `npm start` or `yarn start`.
-      To create a production bundle, use `npm run build` or `yarn build`.
-    -->
+    <div class="video-background" style="background-color: #000">
+      <iframe
+        width="100%"
+        height="100%"
+        src="https://www.youtube.com/embed/RP4lbSymeZ4?start=215&disablekb=1&controls=0&autoplay=1&mute=1&rel=0&loop=1&playlist=RP4lbSymeZ4"
```
Comment (jchan18) [inline client/public/index.html:40]: might need some accessibility changes for the header colors with this specific video, but eh problem for later haha

File: client/src/App.js [hunk 2]
```diff
@@ -19,26 +20,21 @@ import { MEMBERS, ERAS, TYPES } from 'utils';
...
-  const setBackgroundImage = () => {
-    const location = window.location.pathname
-    if (location === '/' || location === '/archive') {
-      return 'image'
-    }
-    return null
-  };
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
```
Comment (jchan18) [inline client/src/App.js:28]: im a little confused, when will this trigger if there are no dependencies? iirc, it'll just run once, but will we ever see the loading state then?

File: client/src/App.js [hunk 3]
```diff
@@ -48,19 +44,19 @@ const App = () => {
           </Route>
           <Route path="/videos/members/:name" component={MemberPage} />
           <Route path="/videos/members">
-            <ByCategory category="member" categoryList={MEMBERS}/>
+            <ByCategory category="member" categoryList={MEMBERS} />
```
Comment (jchan18) [inline client/src/App.js:47]: lol is this somethign we can add to the prettier stuff to do automatically

File: client/src/components/CategoryCard.js [hunk 4]
```diff
@@ -2,23 +2,60 @@ import React from 'react';
...
+    to left bottom,
+    rgba(0, 0, 0, 0.8) 71%,
+    rgba(0, 0, 0, 0.8) 86%,
+    rgba(0, 0, 0, 0.9)
+  );
+  border-radius: inherit;
 
   &:hover {
-    box-shadow: rgba(0, 0, 0, 0.2) 0 0 10px, rgba(255, 255, 255, 0.2) 0 0 10px;
+    -webkit-mask-image: unset;
```
Comment (jchan18) [inline client/src/components/CategoryCard.js:34]: ngl i dont know what half these css things do, but i trust you do :D

File: client/src/components/CategoryCard.js [hunk 5]
```diff
@@ -39,9 +76,16 @@ const CategoryCard = ({ name }) => {
   // TODO: find images to place in the cards
```
Comment (jchan18) [inline client/src/components/CategoryCard.js:76]: can probably remove this todo now

File: client/src/components/GenericVideoPlayer.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+const CurrentVideo = styled.div`
+  width: 100%;
+  text-align: center;
+  margin-bottom: 8%;
+`;
+
+const GenericVideoPlayer = ({ video, options, setCurrentVideo }) => {
+  const { video_id, provider, title } = video;
+
+  if (provider === 'dailymotion') {
```
Comment (jchan18) [inline client/src/components/GenericVideoPlayer.js:36]: It looks liek the stuff inside <CurrentVideo> is pretty much the same, whaat if we just do
```
return (
    <CurrentVideo>
      <InlineDiv>
        <h2>{title}</h2>
        <Button onClick={() => setCurrentVideo(null)}>Clear Video</Button>
      </InlineDiv>
      {provider==='dailymotion' ? <Dailymotion ... />  : <YouTube ... />}
    </CurrentVideo>
  );
```

File: client/src/components/LoadingState.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  width: 100%;
+  position: absolute;
+  top: 35%;
+  text-align: center;
+`;
+
+const LoadingState = () => {
+  return (
+    <Loading>
+      <img alt="loading" src={`${CDN_ASSET_BASE}/assets/loading-heart.gif`} />
```
Comment (jchan18) [inline client/src/components/LoadingState.js:16]: this sounds so cute, what does the loading heart look like LOL

**Review by jchan18 — APPROVED** [review]
Looks awesome!!!! I left some suggestions, but it's not blocking :)

File: client/public/index.html (hunk 1 above)
Comment (mjjcha) [reply client/public/index.html:40]: Yeah, the links probably have to be made a lighter shade of grey, but something for later haha

File: client/src/App.js (hunk 2 above)
Comment (mjjcha) [reply client/src/App.js:28]: Yeah, it basically gets called when the component is mounted. We would be able to see the loading state every time the route changed, but only if we wrap `setIsLoading` with the `setTimeout` - right now, it loads way too quickly and the loading state doesn't even flash lol. I was kinda playing around with this cuz revelsubs had their animated svg, but it does seem kind of useless/annoying :thinking:

File: client/src/App.js (hunk 3 above)
Comment (mjjcha) [reply client/src/App.js:47]: Yup! I have it configured on my end to format on save

File: client/src/components/CategoryCard.js (hunk 4 above)
Comment (mjjcha) [reply client/src/components/CategoryCard.js:34]: haha I was playing with applying a linear gradient on the category cards - it doesn't look that great :grimacing:

File: client/src/components/GenericVideoPlayer.js (hunk 6 above)
Comment (mjjcha) [reply client/src/components/GenericVideoPlayer.js:36]: Good call! At first, I thought I'd have to apply different styling to the wrapper div, but it looks like both can share the same one :D

File: client/src/components/LoadingState.js (hunk 7 above)
Comment (mjjcha) [reply client/src/components/LoadingState.js:16]: It looks like this

**Pull request merged**